src/core/common/settings.hpp

```diff
@@ -28,94 +28,430 @@
+     *
+     */
+    struct NetworkInfo
+    {
+        uint8_t         mRole;                   ///< Current Thread role.
+        uint8_t         mDeviceMode;             ///< Device mode setting.
+        uint16_t        mRloc16;                 ///< RLOC16
+        uint32_t        mKeySequence;            ///< Key Sequence
+        uint32_t        mMleFrameCounter;        ///< MLE Frame Counter
+        uint32_t        mMacFrameCounter;        ///< MAC Frame Counter
+        uint32_t        mPreviousPartitionId;    ///< PartitionId
+        Mac::ExtAddress mExtAddress;             ///< Extended Address
+        uint8_t         mMlIid[OT_IP6_IID_SIZE]; ///< IID from ML-EID
+    };
 
-/**
- * This structure represents the parent information for settings storage.
- *
- */
-struct ParentInfo
-{
-    Mac::ExtAddress mExtAddress; ///< Extended Address
-};
+    /**
+     * This structure represents the parent information for settings storage.
+     *
+     */
+    struct ParentInfo
+    {
+        Mac::ExtAddress mExtAddress; ///< Extended Address
+    };
 
-/**
- * This structure represents the child information for settings storage.
- *
- */
-struct ChildInfo
-{
-    Mac::ExtAddress mExtAddress; ///< Extended Address
-    uint32_t        mTimeout;    ///< Timeout
-    uint16_t        mRloc16;     ///< RLOC16
-    uint8_t         mMode;       ///< The MLE device mode
+    /**
+     * This structure represents the child information for settings storage.
+     *
+     */
+    struct ChildInfo
+    {
+        Mac::ExtAddress mExtAddress; ///< Extended Address
+        uint32_t        mTimeout;    ///< Timeout
+        uint16_t        mRloc16;     ///< RLOC16
+        uint8_t         mMode;       ///< The MLE device mode
+    };
+
+    /**
+     * This constructor initializes the `Settings` object.
+     *
+     * @param[in]  aInstance     A reference to the OpenThread instance.
+     *
+     */
+    explicit Settings(Instance &aInstance)
+        : InstanceLocator(aInstance)
+    {
+    }
+
+    /**
+     * This method initializes the platform settings (non-volatile) module.
+     *
+     * This should be called before any other method from this class.
+     *
+     */
+    void Init(void);
+
+    /**
+     * This method removes all settings from the non-volatile store.
+     *
+     */
+    void Wipe(void);
+
+    /**
+     * This method saves the Operational Dataset (active or pending).
+     *
+     * @param[in]   aIsActive   Indicates whether Dataset is active or pending.
+     * @param[in]   aBuffer     A byte buffer containing the dataset.
+     * @param[in]   aSize       The size of buffer (number of bytes).
+     *
+     * @retval OT_ERROR_NONE              Successfully saved Dataset.
+     * @retval OT_ERROR_NOT_IMPLEMENTED   The platform does not implement settings functionality.
+     *
+     */
+    otError SaveDataset(bool aIsActive, const uint8_t *aBuffer, uint16_t aSize)
```

Comment:
May be better to call this `OperationalDataset` rather than just `Dataset` since the Thread specification defines other kinds of datasets as well.

Reply:
In new pushed commit, changed the method names. Also changed the param to be a `Dataset` reference instead of buffer pointers.